crates/holochain/src/conductor/api/api_external/app_interface.rs

```diff
@@ -139,6 +140,29 @@ impl AppInterfaceApi {
                     .provide_memproofs(&installed_app_id, memproofs)
                     .await?;
                 Ok(AppResponse::Ok)
+            }
+            AppRequest::EnableAfterMemproofsProvided => {
+                let status = self
+                    .conductor_handle
+                    .get_app_info(&installed_app_id)
+                    .await?
+                    .ok_or(ConductorApiError::other("app not found".to_string()))?
+                    .status;
+                match status {
+                    AppInfoStatus::Running
```

Comment:
I like not erroring here but it should be a no-op? Could this be a separate match branch that just returns `Ok(AppResponse::Ok)`

Reply:
I prefer letting it go through to the state transition function, since that already comprehensively handles every combination of previous and next state, and defines what is a no-op and not. Then if we want to actually cause some change when Enabling while already Running, we can do so without hunting down places where we bypassed that.

Reply:
no-op specified here: https://github.com/holochain/holochain/blob/b3c7f90435c70922979f75d76bb8761ef7a0957f/crates/holochain_types/src/app.rs#L882